src/vmm/src/devices/virtio/net/device.rs

```diff
@@ -644,6 +641,18 @@ impl Net {
         }
     }
 
+    fn build_tap_supported_features(desired_flags: Vec<(u32, u32)>, supported_features: u64) -> u32 {
+        let mut answer : u32 = 0;
+
+        for i in 0..desired_flags.len() {
+            if (supported_features & 1 << desired_flags[i].1) != 0 {
+                answer |= desired_flags[i].0
+            }
+        }
```

Comment:
This can be:
```rust
fn build_tap_supported_features(desired_flags: &[(u32, u32)], supported_features: u64) -> u32 {
     let mut answer: u32 = 0;

   for (tap_flag, net_flag) in desired_flags.iter() {
       if (supported_features & 1 << net_flag) != 0 {
           answer |= tap_flag;
       }
   }

   answer
}
```
Basically if you pass `Vec<(u32, u32)>` then the function takes ownership of the `desired_flags`. This means if you call this function multiple times you will need to create new instance if `desired_flags` each time.
To convert from `Vec<(u32, u32)>` to `&[(u32, u32)]` you can simply use `&` before the vector on the calling side.

Also you can simply iterate over a vector of tuples, instead of accessing them through indexes.

Reply:
thx. done